### PR TITLE
[vsock] remove unused feature to reduce deps

### PR DESCRIPTION
**Pull request by ikicha**
### Summary of the PR

this project uses only yaml config, so remove some of features in config to reduce deps

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [ V ] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [ V ] All added/changed functionality has a corresponding unit/integration
  test.
- [ V ] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [ V ] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by stefano-garzarella — APPROVED** [review]

**Comment by ikicha** [comment]
@vireshk , can you review this PR? It looks like it needs 2 approvals to merge this.

**Comment by vireshk** [comment]
@ikicha can you please rebase this again, there are some conflicts ?

**Comment by ikicha** [comment]
> @ikicha can you please rebase this again, there are some conflicts ?

Done!

File: Cargo.lock [hunk 1]
```diff
@@ -1246,9 +1043,9 @@ dependencies = [
 
 [[package]]
 name = "vmm-sys-util"
-version = "0.11.1"
+version = "0.11.2"
```
Comment (stefano-garzarella) [inline Cargo.lock:1046]: Why `vmm-sys-util` is upgrading, I don't think it is a dependency of `config` or `serde`.

Did you run a generic `cargo update`?

Comment (ikicha) [reply Cargo.lock]: Yeah, I had. I uploaded the another patchset with just cargo build.

**Review by vireshk — APPROVED** [review]

**Pull request merged**